### PR TITLE
(Pilot) Create a common goal for 259 grants across regions

### PR DESCRIPTION
**Pull request by kryswisnaskas**
## Description of change
This PR contains a script that will create a template and a goal for grants/recipients listed in the CSV file.

## How to test
Make sure tests pass. 

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1340

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [n/a] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [n/a] API Documentation updated
- [n/a] Boundary diagram updated
- [n/a] Logical Data Model updated
- [n/a] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [n/a] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: tests/activity-report.spec.ts [hunk 1]
```diff
@@ -397,4 +397,16 @@ test.describe("Activity Report", () => {
...
+    // create a new report
+    await page.getByRole('link', { name: 'Activity Reports' }).click();
+    await page.getByRole('button', { name: '+ New Activity Report' }).click();
+
+    // select a recipient
+    await page.getByRole('group', { name: 'Was this activity for a recipient or other entity? *' }).locator('label').filter({ hasText: 'Recipient' }).click();
+    await page.locator('#activityRecipients div').filter({ hasText: '- Select -' }).nth(1).click();
+    await page.locator('#react-select-3-option-0-0').click();
+  });
 });
```
Comment (kryswisnaskas) [inline tests/activity-report.spec.ts:412]: This is not part of this PR,  but got included as a result of fixing an incorrect merge.

File: src/tools/goalPilot.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      where: { templateName: goalName },
+      defaults: { templateName: goalName },
+    });
+
+    for await (const el of recipients) {
+      let number = '';
+
+      number = el['Grant Number'];
+
+      const grant = await Grant.findOne(
```
Comment (thewatermethod) [inline src/tools/goalPilot.js:52]: do we need to explicitly handle "grant not found" case? I understand we'll get a ```can't read property id of undefined``` error and go to the catch, but if we throw an if statement in here, we could proceed through the list creating the rest of the goals. Maybe it doesn't matter because we have a list with presumably correct data on it.

**Review by thewatermethod — APPROVED** [review]
I had issues running locally but the tests pass and I assume this was tested on dev

**Review by GarrettEHill — DISMISSED** [review]
While this works, I think this is the wrong path to accomplish the aspiration of allowing the same goal to be assessable by a large number of recipients.

File: package.json [hunk 3]
```diff
@@ -74,6 +74,8 @@
     "import:reports:local": "./node_modules/.bin/babel-node ./src/tools/importSSActivityReports.js",
     "import:reports": "node ./build/server/tools/importSSActivityReports.js",
     "import:goals:local": "./node_modules/.bin/babel-node ./src/tools/importTTAPlanGoals.js",
+    "create:goals": "node ./build/server/tools/goalPilotCLI.js",
```
Comment (nvms) [inline package.json:77]: with the introduction of the typescript compiler, the build paths have changed slightly, so this path will not work once merged. it should be updated to `./build/server/src/tools/goalPilotCLI.js`:

Screen Shot 2023-01-25 at 11 12 08 AM

**Review by AdamAdHocTeam — APPROVED** [review]
Looks good but need to get test passing.

File: src/tools/goalPilot.js (hunk 2 above)
Comment (kryswisnaskas) [reply src/tools/goalPilot.js:53]: Added a more graceful handling of the cases where a grant can't be found.

File: package.json [hunk 4]
```diff
@@ -74,6 +74,8 @@
...
+    "create:goals:local": "./node_modules/.bin/babel-node ./src/tools/goalPilotCLI.js",
```
[the remark this answers is not in the document]
Comment (kryswisnaskas) [reply package.json:78]: Glad to hear this is no longer an issue.

**Comment by kryswisnaskas** [comment]
> While this works, I think this is the wrong path to accomplish the aspiration of allowing the same goal to be assessable by a large number of recipients.

Discussed this.  Currated templates will be implemented in the next phase when we have the groups to template connection.

File: package.json (hunk 3 above)
Comment (kryswisnaskas) [reply package.json]: Thanks. Adjusted.
